src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -350,7 +350,8 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                             {
                                 // Remove the base directory prefix (including trailing / that OS X adds) and 
                                 // add the paired event to the list of events to skip and notify the user of the rename
-                                string newPathRelativeName = eventPaths[pairedId].Remove(0, _fullDirectory.Length + 1);
+                                int removeLength = _fullDirectory[_fullDirectory.Length - 1] == '/' ? _fullDirectory.Length : _fullDirectory.Length + 1; 
```

Comment:
Rather than doing this check every time, can we instead normalize _fullDirectory when it's created to just always include the trailing slash?